src/subcommand/server/account.rs

```diff
@@ -0,0 +1,226 @@
+use crate::subcommand::server::database::Database;
+use crate::subcommand::server::error::ServerResult;
+use axum::extract::Path;
+use axum::http::StatusCode;
+use axum::response::{IntoResponse, Response};
+use axum::routing::{get, post};
+use axum::{Extension, Json, Router};
+use bip322::verify_simple;
+use bitcoin::consensus::Decodable;
+use bitcoin::{Address, Witness};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub(crate) struct Account {
+    pub(crate) btc_address: String,
+    pub(crate) ln_address: String,
+    pub(crate) past_ln_addresses: Vec<String>,
+    pub(crate) total_diff: i64,
+    pub(crate) payouts: Vec<HistoricalPayout>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub(crate) struct HistoricalPayout {
+    pub(crate) amount: u32,
+    pub(crate) allocated_diff: i64,
+    pub(crate) block_start: u32,
+    pub(crate) block_end: u32,
+    pub(crate) status: String,
+    pub(crate) failure_reason: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub(crate) struct AccountLookup {
+    pub(crate) btc_address: String,
+    pub(crate) signature: Option<String>,
+    pub(crate) nonce: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub(crate) struct AccountUpdate {
+    pub(crate) btc_address: String,
+    pub(crate) ln_address: String,
+    pub(crate) signature: String,
+    pub(crate) nonce: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub(crate) struct AccountResponse {
+    pub(crate) success: bool,
+    pub(crate) remark: Option<String>,
+}
+
+pub fn account_router() -> Router {
+    Router::new()
+        .route("/account/{address}", get(account_lookup))
+        .route("/account/details", post(account_detail))
+        .route("/account/update", post(account_update))
```

Comment:
In order to simplify (we can iterate on this later) I would suggest we just have two endpoints.

The /account/{address} that simply returns the the btc address, ln_address and past ln addresses (no share data or payouts.

Change /account/update to /account/{address} but keeping the post which allows to update the ln address if a signature on just the string of the ln address with the {address} is provided.

```suggestion
        .route("/account/{address}", get(account_lookup))
         .route("/account/{address}", post(account_update))
```

Reply:
nack, endpoints are already implemented and separation of concerns should simplify front-end development by providing primary information on a simpler GET endpoint that requires nothing except an address.